packages/shared/src/lib/auxiliary/wallet-connect/utils/switchToRequiredAccount.ts

```diff
@@ -0,0 +1,28 @@
+import { IAccountState } from '@core/account'
+import { getSelectedAccountIndex } from '@core/account/stores'
+import { IChain } from '@core/network'
+import { findActiveAccountWithAddress } from '@core/profile/actions'
+import { openPopup, PopupId } from '../../../../../../desktop/lib/auxiliary/popup'
+
+export function switchToRequiredAccount(accountAddress: string, chain: IChain): Promise<IAccountState> {
+    return new Promise((resolve, reject) => {
+        const account = findActiveAccountWithAddress(accountAddress, chain.getConfiguration().id)
+        if (!account) {
+            reject('Could not find address')
```

Comment:
maybe to fix the problem from my other comment, we could `resolve(undefined)` here and check for that in the parent?

Reply:
I tried it out, but I didn't like it.